lib/mocha/stubbed_method.rb

```diff
@@ -37,20 +35,7 @@ def reset_mocha
     def hide_original_method
       return unless original_method_owner.__method_exists__?(method_name)
       store_original_method_visibility
-      if use_prepended_module_for_stub_method?
-        use_prepended_module_for_stub_method
-      else
-        begin
-          store_original_method
-        # rubocop:disable Lint/HandleExceptions
-        rescue NameError
-          # deal with nasties like ActiveRecord::Associations::AssociationProxy
-        end
-        # rubocop:enable Lint/HandleExceptions
-        if stub_method_overwrites_original_method?
-          remove_original_method_from_stubbee
-        end
```

Comment:
TODO: lots more cleaning up required here, I'm not sure if I'm missing something as well, as tests are being modified in tandem. Probably safer to do smaller removals.

Reply:
> Probably safer to do smaller commits.

Yes, I think it would be. e.g. hard-code `Mocha:: RUBY_V2_PLUS` to return `true` in one commit and then gradually collapse the associated call-sites with one change per commit...?